drivers/cache/cache_handlers.c

```diff
@@ -7,30 +7,32 @@
 #include <zephyr/cache.h>
 #include <zephyr/syscall_handler.h>
 
-static inline int z_vrfy_sys_cache_data_all(int op)
+static inline int z_vrfy_sys_cache_data_flush_all(void)
 {
-	return z_impl_sys_cache_data_all(op);
+	return z_impl_sys_cache_data_flush_all();
 }
-#include <syscalls/sys_cache_data_all_mrsh.c>
+#include <syscalls/sys_cache_data_flush_all_mrsh.c>
 
-static inline int z_vrfy_sys_cache_data_range(void *addr, size_t size, int op)
+static inline int z_vrfy_sys_cache_data_flush_range(void *addr, size_t size)
 {
 	Z_OOPS(Z_SYSCALL_MEMORY_WRITE(addr, size));
 
-	return z_impl_sys_cache_data_range(addr, size, op);
+	return z_impl_sys_cache_data_flush_range(addr, size);
```

Comment:
This seems fine to expose as a usermode syscall. I'm not so sure about the flush all, as harmless as it may seem.

Reply:
yeah, I replied above.